netket/hilbert/abstract_hilbert.py

```diff
@@ -32,6 +32,18 @@ class NoneType:
     pass
 
 
+legacy_warn_str = (
+    "This choice of arguments for `hilbert.random_state` is deprecated and "
+    "will be removed in a future version.\n"
+    "The new syntax mimicks jax.random.normal, and is:\n"
+    "    hilbert.random_state(prngkey, size=1, dtype=jnp.float32)\n"
+    "the first argument is mandatory and must be a valid jax PRNGKey."
```

Comment:
I think "mimics `jax.random.normal` is no quite true, maybe this is a bit more clear?
```suggestion
    "The new syntax is\n"
    "    hilbert.random_state(prngkey, size=1, dtype=jnp.float32)\n"
    "and, like in `jax.random`, the first argument is mandatory and must be a valid jax `PRNGKey`."
```

Reply:
I mean, the arguments are literally the same, `key, shape, dtype` (though for us shape is size)